diff --git a/node_modules/d3-transition/src/transition/end.js b/node_modules/d3-transition/src/transition/end.js
index 252ce86..4b2040d 100644
--- a/node_modules/d3-transition/src/transition/end.js
+++ b/node_modules/d3-transition/src/transition/end.js
@@ -2,8 +2,9 @@ import {set} from "./schedule.js";
 
 export default function() {
   var on0, on1, that = this, id = that._id, size = that.size();
+  const rejectArg = this;
   return new Promise(function(resolve, reject) {
-    var cancel = {value: reject},
+    var cancel = {value: () => reject(rejectArg)},
         end = {value: function() { if (--size === 0) resolve(); }};
 
     that.each(function() {
